zookeeper-server/src/main/java/org/apache/zookeeper/client/FourLetterWordMain.java

```diff
@@ -137,7 +195,9 @@ public static String send4LetterWord(
                 sock = new Socket();
                 sock.connect(hostaddress, timeout);
             }
+            sock.setSoLinger(soLingerOn, soLinger);
             sock.setSoTimeout(timeout);
+            sock.setTcpNoDelay(tcpNoDelay);
```

Comment:
```suggestion
            sock.setSoLinger(false, -1);
            sock.setSoTimeout(timeout);
            sock.setTcpNoDelay(true);
```

Sorry for my distractive review comments. I means these two options are suitable for `send4LetterWord` for reasons:
1. `send4LetterWord` is one-shot, it is better to send request immediately and return immediately once got response to not block caller.
2. `sock.setTcpNoDelay(true)` tells kernel to not delay any send operations.
3. `sock.setSoLinger(false, -1)` tells kernel to not block `close`.

I think there is little chance for people to configure them if any.